### PR TITLE
test(database): add e2e case of declarative database with delete reclaim policy

### LINKED ISSUE #5680: [split from `#4679`]: Improve and finalize E2E coverage
- E2E test also missing a test with retain policy of Delete:
  - we can create a new CRD to re-adopt the existing DB, but with retain: delete
  - then delete the CRD
- E2E new feature_type for declarative databases
- E2E test what happens when we delete the namespace or the CNPG cluster

### PR DESCRIPTION
**Pull request by gabriele-wolfox**
Closes #5680

### CODE REVIEW

File: tests/e2e/declarative_database_management_test.go [hunk 1]
```diff
@@ -48,18 +51,20 @@ var _ = Describe("Declarative databases management test", Label(tests.LabelSmoke
...
+			databaseWithDeleteRetainPolicy  *apiv1.Database
+			err                             error
 		)
 
 		BeforeAll(func() {
-			// Create a cluster in a namespace we'll delete after the test
-			namespace, err = env.CreateUniqueTestNamespace(namespacePrefix)
+			// We set the namespace name as a constant, as we are going to test the namespace deletion inside
+			// the suite (not after), in order to check the finalizers removal
+			err = env.CreateNamespace(namespace)
```
Comment (jsilvela) [inline tests/e2e/declarative_database_management_test.go:67]: why not use the CreateUnique function?

File: tests/e2e/declarative_database_management_test.go [hunk 2]
```diff
@@ -154,13 +159,83 @@ var _ = Describe("Declarative databases management test", Label(tests.LabelSmoke
...
+
+				By("verifying the db exists", func() {
+					primaryPodInfo, err := env.GetClusterPrimary(namespace, clusterName)
+					Expect(err).ToNot(HaveOccurred())
+
+					assertDatabaseExists(namespace, primaryPodInfo.Name, dbname, true)
+				})
+			})
+
+			It("maintains the db when Database CRD is removed", func() {
```
Comment (jsilvela) [inline tests/e2e/declarative_database_management_test.go:200]: wait ... this is exactly the opposite. Inside it's checking the Postgres DB does NOT exist, line 209

Comment (gabriele-wolfox) [reply tests/e2e/declarative_database_management_test.go:200]: You are right, it should be "does not maintain..."

File: tests/e2e/declarative_database_management_test.go [hunk 3]
```diff
@@ -101,65 +100,147 @@ var _ = Describe("Declarative databases management test", Label(tests.LabelSmoke
...
-					assertDatabaseExists(namespace, primaryPodInfo.Name, dbname, true)
-				})
+				Eventually(func(g Gomega) {
+					err := env.Client.Get(env.Ctx, databaseNamespacedName, dbObj)
+					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(dbObj.Status.Ready).Should(BeTrue())
+				}, 300).WithPolling(10 * time.Second).Should(Succeed())
+			})
+			By("deleting the namespace and making sure it succeeds before timeout", func() {
+				err := env.DeleteNamespaceAndWait(namespace, 30)
```
Comment (jsilvela) [inline tests/e2e/declarative_database_management_test.go:242]:
```suggestion
				err := env.DeleteNamespaceAndWait(namespace, 60)
```

**Review by jsilvela — APPROVED** [review]

File: tests/e2e/declarative_database_management_test.go (hunk 3 above)
Comment (gabriele-wolfox) [reply tests/e2e/declarative_database_management_test.go]: Yep, I started with 60 seconds, but after exercising the suite I guess it is fine to reduce it.

**Review by NiccoloFei — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Pull request merged**